app/Services/PaymentService.php

```diff
@@ -38,7 +38,15 @@ public static function logPayment($data, $model)
     public static function logLateFee($data)
     {
         try {
-            $payment = LateFee::updateOrCreate(['id' => $data['id']], ['date_paid' => $data['date_paid']]);
+            if (isset($data['id'])) {
+
+                $payment = LateFee::updateOrCreate(['id' => $data['id']], ['date_paid' => $data['date_paid'], 'amount' => $data->amount]);
+            } else {
+                // dd('i got here');
+
+                $payment = LateFee::updateOrCreate($data);
```

Comment:
why are we re-running lines 47 and 49 after 43?

Reply:
it is working for both the creating late-fee and updating late fee

Reply:
mistake